torch/nn/parallel/replicate.py

```diff
@@ -56,6 +56,29 @@ def descendant_modules(module):
     return True
 
 
+def _to_device_index(devices):
+    if len(devices) <= 0:
+        raise RuntimeError("Cannot replicate using an empty device list.")
+
+    if isinstance(devices, list) and isinstance(devices[0], list):
+        device_ids = []
+        for i, device in enumerate(devices):
+            assert len(device) == len(devices[0]), (
+                "Cannot replicate to unidentical number of devices, but got "
+                "device list {} and {} for replica {} and {}."
+            ).format(devices[0], devices[i], 0, i)
+
+            assert len(device) == len(set(device)), (
+                "Duplicated device ids {} for replica {}."
+            ).format(device, i)
```

Comment:
Do you also want to check there are no duplicates across replicas?

Theoretically it would be fine to use `[[0,1], [1,2]]`, but I would expect it to be an error 99% of the time...

Reply:
I use to have that, but then I realized I cannot test it with 2 GPUs, so I removed it.

Reply:
check added